metrics.nim

```diff
@@ -439,35 +462,47 @@ template declarePublicGauge*(identifier: untyped,
 
 proc incGauge(gauge: Gauge, amount: int64|float64 = 1, labelValues: LabelsParam = @[]) =
   when defined(metrics):
-    var timestamp = getTime().toMilliseconds()
+    try:
+      var timestamp = getTime().toMilliseconds()
 
-    let labelValuesCopy = validateGaugeLabelValues(gauge, labelValues)
+      let labelValuesCopy = validateGaugeLabelValues(gauge, labelValues)
 
-    atomicAdd(gauge.metrics[labelValuesCopy][0].value.addr, amount.float64)
-    atomicStore(cast[ptr int64](gauge.metrics[labelValuesCopy][0].timestamp.addr), timestamp.addr, ATOMIC_SEQ_CST)
+      atomicAdd(gauge.metrics[labelValuesCopy][0].value.addr, amount.float64)
+      atomicStore(cast[ptr int64](gauge.metrics[labelValuesCopy][0].timestamp.addr), timestamp.addr, ATOMIC_SEQ_CST)
 
-    pushMetrics(name = gauge.name,
-                value = gauge.metrics[labelValuesCopy][0].value,
-                metricType = "g",
-                timestamp = timestamp)
+      pushMetrics(name = gauge.name,
+                  value = gauge.metrics[labelValuesCopy][0].value,
+                  metricType = "g",
+                  timestamp = timestamp)
+    except Exception as e:
+      try:
+        write stderr, "metrics: " & e.msg
+      except IOError:
+        discard
```

Comment:
Same as above. This library should not encourage erroneous API usage.

Reply:
it's not erroneous usage - it's unknown errors coming out of the bowels of the library from code we don't trust

Reply:
What bowels? What code?

Reply:
ask Nim - it has a `raises Exception` effect coming out of the standard library